src/cpp/FilesInformation/Loaders/FilesByNameLoader.cpp

```diff
@@ -127,15 +128,60 @@ TEST_CASE("Empty directory") {
     std::filesystem::remove("tmp");
 }
 
-TEST_CASE("Directory with files") {
+TEST_CASE("Directory with directory") {
     std::filesystem::create_directory("tmp");
     std::filesystem::create_directory("tmp/tmp");
 
     FilesByNameLoader loader;
     Directory* directory = loader.loadDirectory("tmp");
-    CHECK(directory->size() != 0);
+    CHECK(directory->size() == 1);
+    CHECK(dynamic_cast<Directory*>((*directory)[0]) != nullptr);
+
+    delete directory;
+    std::filesystem::remove("tmp/tmp");
+    std::filesystem::remove("tmp");
+}
+
+TEST_CASE("Directory with file") {
+    std::filesystem::create_directory("tmp");
+    std::fstream file("tmp/tmp.txt", std::ios_base::out);
+
+    FilesByNameLoader loader;
+    Directory* directory = loader.loadDirectory("tmp");
+    CHECK(directory->size() == 1);
+    CHECK(dynamic_cast<File*>((*directory)[0]) != nullptr);
+
     delete directory;
+    file.close();
+    std::filesystem::remove("tmp/tmp.txt");
+    std::filesystem::remove("tmp");
+}
+
+TEST_CASE("Directory with file and directory") {
+    std::filesystem::create_directory("tmp");
+    std::filesystem::create_directory("tmp/tmp");
+    std::fstream file("tmp/tmp.txt", std::ios_base::out);
+
+    FilesByNameLoader loader;
+    Directory* directory = loader.loadDirectory("tmp");
+    CHECK(directory->size() == 2);
+    CHECK(dynamic_cast<File*>((*directory)[0]) != nullptr);
```

Comment:
This test is unreliable - unless we enforce order of children, we can't be sure if file will be at index 0 or at index 1
This should be already fixed in 0b5fbe1f3965df6d1a6f48ad26e1db5d703fad84 - but maybe we want some other solution?